rosbag2/include/rosbag2/converter.hpp

```diff
@@ -60,11 +62,13 @@ class ROSBAG2_PUBLIC Converter
   std::shared_ptr<SerializedBagMessage>
   convert(std::shared_ptr<const SerializedBagMessage> message);
 
+  void add_topic(const std::string & topic, const std::string & type);
+
 private:
   std::shared_ptr<SerializationFormatConverterFactoryInterface> converter_factory_;
   std::unique_ptr<SerializationFormatConverterInterface> input_converter_;
   std::unique_ptr<SerializationFormatConverterInterface> output_converter_;
-  std::map<std::string, std::string> topics_and_types_;
+  std::map<std::string, ConverterTypeSupport> topics_and_types_;
```

Comment:
does this have to be an ordered map? What about `std::unordered_map`?

Reply:
Correct a `unordered_map` is sufficient. This is true for almost all our usages of `std::map`. As this refactoring will touch a lot of files I did it as separate pr #65.